### PR TITLE
homework_2

### PR DESCRIPTION
**Pull request by boodscode237**
Dataset can be founded in Kaggle

### CODE REVIEW

**Comment by boodscode237** [comment]
Это домашнее задание по Machine Learning For analysis of scientific data. я не знал как его задать на проверку. я сделал все в срок и вот такая структура. первое домашнее задание на главной ветке второе домашнее задание  на ветке homework_2 он находится в папке notebooks.

**Review by pacifikus — CHANGES_REQUESTED** [review]
Добрый день!

Оставлю комментарии сразу ко всем ДЗ, чтобы было целостное понимание

Комментарии по ДЗ 1:

- Обоснование для метрики нужно приводить в самом начале до проведения экспериментов, пока получается, что вы измеряли в моделях accuracy, считали ее на кросс-валидации, подбирали гиперпараметры, а потом в конце неожиданно говорите, что смотреть надо на recall... Получается, все прошлые эксперименты бесполезны с точки зрения новой метрики...
- Недостает комментариев к графикам - что на них можно увидеть? Почему вы решили посмотреть именно эти распределения/признаки/отношения?

В целом в этом дз не требовалось обучать модели, но раз эта часть есть, оставлю несколько комментариев:
- В `train_test_split` тоже нужно фиксировать `random_state` в аргументах
- В `RandomForestClassifier` также нужно фиксировать `random_state`
- `RandomizedSearchCV` также нужно фиксировать `random_state`

И в целом эта работа почти полностью повторяет [ноутбук](https://github.com/Aahra/End-to-End-Heart-Disease-classification-Milestone-Project-1-/blob/main/End_to_end_Heart_disease_classification_.ipynb)

Комментарии по ДЗ 2:

- Не очень понятно, почему меняется датасет в ДЗ 2 - предполагалось, что на протяжении всех трех ДЗ будет использоваться один и тот же набор данных
- Опять же тогда всплывают моменты из ДЗ 1 - почему мы используем тут именно RMSLE? 
- В этой работе по заданию нужно использовать какую-то "простую" модель, то есть не ансамбль, а RandomForestRegressor - как раз ансамбль

И еще эта работа почти полностью повторяет [ноутбук](https://www.kaggle.com/code/amalap862001/end-to-end-bulldozer-price-regression-project)

**Comment by boodscode237** [comment]
Я переделаю его, используя первый набор данных(heart-disease). Для второго набора данных в конкурсе было указано использовать именно его.

**Comment by boodscode237** [comment]
И еще, код, который я отправлял ранее, был примером курса Udemy, который я прохожу.
